### PR TITLE
EES-3001 Prevent going back/next on table tool before current change completes

### PR DESCRIPTION
**Pull request by ntsim**
This PR:

- Adds new checks to `Wizard` and its associated components to prevent steps changing until the the next/previous step is actually ready.
- Adds loading states via disabling buttons and adding appropriate loading spinners whilst transitioning to next/previous steps.

## Relevant changes

- Refactors out a new `WizardStepSummary` component to remove some of the boilerplate markup and functionality from the read-only views of the wizard steps.
- Various cleanup to tidy up across `Wizard` and associated components.

### CODE REVIEW

File: src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx [hunk 1]
```diff
@@ -222,6 +222,7 @@ describe('FiltersForm', () => {
   const testWizardStepProps: InjectedWizardProps = {
     currentStep: 1,
     isActive: true,
+    isEnabled: true,
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx:225]: Is there any test covering the scenario that isEnabled is false?

File: src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/Wizard.test.tsx [hunk 2]
```diff
@@ -473,10 +474,55 @@ describe('Wizard', () => {
...
+      expect(handleBack).toHaveBeenCalledTimes(1);
+    });
+
+    const step1 = screen.getByTestId('wizardStep-1');
+    const step2 = screen.getByTestId('wizardStep-2');
+
+    expect(step1).not.toHaveAttribute('aria-current');
+    expect(step2).toHaveAttribute('aria-current', 'step');
+
+    jest.advanceTimersByTime(500);
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/Wizard.test.tsx:658]: Nice!!

File: src/explore-education-statistics-common/src/modules/table-tool/components/WizardStepSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            <span className="govuk-visually-hidden">{`on step ${stepNumber}`}</span>
+          </ButtonText>
+
+          <LoadingSpinner
+            alert
+            inline
+            hideText
+            loading={isLoading}
+            size="sm"
+            text="Page is loading"
```
Comment (duncan-at-hiveit) [inline src/explore-education-statistics-common/src/modules/table-tool/components/WizardStepSummary.tsx:43]: Would this spinner text be more useful saying something like "Step {stepNumber} is loading"?

File: src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx (hunk 1 above)
Comment (ntsim) [reply src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx:225]: Test cases around this have been added to `Wizard.test.tsx` as this. As it's only consumed by `WizardStepHeading` to toggle some classes, there isn't really any need to test this further.

File: src/explore-education-statistics-common/src/modules/table-tool/components/WizardStepSummary.tsx (hunk 3 above)
Comment (ntsim) [reply src/explore-education-statistics-common/src/modules/table-tool/components/WizardStepSummary.tsx]: Added loading text for the two loading spinners that should read 'Loading step X' / 'Loading next step' / 'Loading previous step'

Comment (duncan-at-hiveit) [reply src/explore-education-statistics-common/src/modules/table-tool/components/WizardStepSummary.tsx]: Great stuff, ta!

File: src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/FiltersForm.test.tsx:225]: Cool.

**Pull request merged**